src/Microsoft.Management.Deployment/PackageManager.idl

```diff
@@ -87,6 +87,66 @@ namespace Microsoft.Management.Deployment
         }
     }
 
+    /// State of the uninstall
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 4)]
+    enum PackageUninstallProgressState
+    {
+        /// The uninstall is queued but not yet active. Cancellation of the IAsyncOperationWithProgress in this 
+        /// state will prevent the package from uninstalling.
+        Queued,
+        /// The uninstall is in progress. Cancellation of the IAsyncOperationWithProgress in this state will not
+        /// stop the installation or the post uninstall steps.
+        Uninstalling,
+        /// The uninstaller has completed and cleanup actions are in progress. Cancellation of the 
+        /// IAsyncOperationWithProgress in this state will not stop cleanup or roll back the uninstall.
+        PostUninstall,
+        /// The operation has completed.
+        Finished,
+    };
+
+    /// Progress object for the uninstall
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 4)]
+    struct UninstallProgress
+    {
+        /// State of the uninstall
+        PackageUninstallProgressState State;
+        /// Uninstall percentage if known.
+        Double UninstallationProgress;
+    };
+
+    /// Status of the uninstall call
+    /// Implementation Note: Errors mapped from AppInstallerErrors.h
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 4)]
+    enum UninstallResultStatus
+    {
+        Ok,
+        BlockedByPolicy,
+        CatalogError,
+        InternalError,
+        InvalidOptions,
+        UninstallError,
+        ManifestError,
+    };
+
+    /// Result of the install
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 4)]
+    runtimeclass UninstallResult

```

Comment:
>UninstallResult [](http://example.com/codeflow?start=17&length=15)

we should make sure when this feature gets checked in we FCIB the DesktopAppInstaller package to SV3, so that dependent callers (backup / restore) are not broken